DylibBinder/DBFunc.cs

```diff
@@ -0,0 +1,141 @@
+	public class DBFunc {
+		public DBFunc (TLFunction tlf, bool isMutating = false, bool isProtocol = false)
+		{
+			Name = tlf.Name.Name;
+			IsStatic = GetStaticStatus (tlf.Signature);
+			IsProperty = "False";
+			HasThrows = tlf.Signature.CanThrow.ToString ();
+			IsMutating = isMutating.ToString ();
+			OperatorKind = tlf.Operator.ToString ();
+			HasInstance = (!tlf.Signature.IsConstructor && IsStatic == "False").ToString ();
+			// TODO not sure how to handle instances with protocols yet
+			if (isProtocol)
+				HasInstance = "False";
+
+			ReturnType = SwiftTypeToString.MapSwiftTypeToString (tlf.Signature.ReturnType, tlf.Module.Name);
+			ParameterLists = new DBParameterLists (tlf.Signature, HasInstance);
+			GenericParameters = new DBGenericParameters (tlf.Signature);
+		}
+
+		public DBFunc (DBProperty dbProperty, string propertyType, bool isMutating = false)
+		{
+			if (propertyType == "Getter")
+				Name = $"get_{dbProperty.Name}";
+			else
+				Name = $"set_{dbProperty.Name}";
+			PropertyType = propertyType;
+			IsStatic = dbProperty.IsStatic;
+			IsProperty = "True";
+			HasThrows = "False";
+			IsMutating = isMutating.ToString ();
+			HasInstance = IsStatic == "True" ? "False" : "True";
+			ReturnType = dbProperty.Type;
+			ParameterLists = new DBParameterLists (ReturnType, HasInstance, propertyType);
+			GenericParameters = dbProperty.GenericParameters;
+		}
+
+		public string Name { get; }
+		public string IsStatic { get; }
+		public string IsProperty { get; }
+		public string ReturnType { get; }
+		public string HasThrows { get; }
+		public string IsMutating { get; }
+		public string HasInstance { get; }
+
+		public string PropertyType { get; } = "";
+		public string IsPossiblyIncomplete { get; } = "False";
+		public string OperatorKind { get; } = "None";
+		public string Accessibility { get; } = "Public";
+		public string IsFinal { get; } = "False";
+		public string IsDeprecated { get; } = "False";
+		public string IsUnavailable { get; } = "False";
+		public string IsOptional { get; } = "False";
+		public string IsRequired { get; } = "False";
+		public string IsConvenienceInit { get; } = "False";
+		public string ObjcSelector { get; } = "";
+
+		public DBParameterLists ParameterLists { get; }
+		public DBGenericParameters GenericParameters { get; }
+		public List<DBAssociatedType> AssociatedTypes { get; } = new List<DBAssociatedType> ();
+
+		string GetStaticStatus (SwiftBaseFunctionType signature)
+		{
+			if (signature is SwiftStaticFunctionType)
+				return "True";
+			else if (signature is SwiftStaticFunctionThunkType)
+				return "True";
+			else if (signature is SwiftPropertyType propType)
+				return propType.IsStatic.ToString ();
+			else if (signature is SwiftPropertyThunkType propThunkType)
+				return propThunkType.IsStatic.ToString ();
+			else if (signature is SwiftUncurriedFunctionThunkType)
+				return "False";
+			else if (signature is SwiftUncurriedFunctionType)
+				return "False";
+
+			return "False";
```

Comment:
Here, return a bool, second, that 'is' check can be written in a more nicer way using [type matching on a switch](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements) from the docs:
```csharp
public static double ComputeAreaModernSwitch(object shape)
{
    switch (shape)
    {
        case Square s:
            return s.Side * s.Side;
        case Circle c:
            return c.Radius * c.Radius * Math.PI;
        case Rectangle r:
            return r.Height * r.Length;
        default:
            throw new ArgumentException(
                message: "shape is not a recognized shape",
                paramName: nameof(shape));
    }
}
```

Reply:
I started this, but I am using C# 8 and it is available in C# 9, it's fine to go to C# 9 right?
Disccusion